diff --git a/src/GearcolecoCore.cpp b/src/GearcolecoCore.cpp
--- a/src/GearcolecoCore.cpp
+++ b/src/GearcolecoCore.cpp
@@ -815,7 +815,7 @@ bool GearcolecoCore::LoadState(std::istream& stream)
 
         Debug("Load state stream size: %d", size);
 
-        GC_SaveState_Header_Libretro header;
+        GC_SaveState_Header_Libretro header = {};
 #if !defined(__LIBRETRO__)
         bool is_desktop_savestate = false;
 #endif
