diff --git a/src/libretro/config/config.cpp b/src/libretro/config/config.cpp
index d1cc06f..3695d2d 100644
--- a/src/libretro/config/config.cpp
+++ b/src/libretro/config/config.cpp
@@ -49,8 +49,10 @@
 #include <SPI.h>
 #include <SPI_Firmware.h>
 #include <FreeBIOS.h>
 #include <file/config_file.h>
+#ifdef HAVE_NETWORKING_DIRECT_MODE
 #include <Net_PCap.h>
+#endif
 
 #include "config/constants.hpp"
 #include "config/definitions.hpp"
diff --git a/src/libretro/format.cpp b/src/libretro/format.cpp
index cb25834..a99cd88 100644
--- a/src/libretro/format.cpp
+++ b/src/libretro/format.cpp
@@ -81,6 +81,7 @@ auto fmt::formatter<MelonDsDs::FormattedGLEnum>::format(MelonDsDs::FormattedGLE
 }
 #endif
 
+#ifdef HAVE_NETWORKING_DIRECT_MODE
 auto fmt::formatter<MelonDsDs::FormattedPCapFlags>::format(MelonDsDs::FormattedPCapFlags e, format_context& ctx) const -> decltype(ctx.out()) {
     auto flags = static_cast<uint32_t>(e);
 
@@ -118,6 +119,7 @@ auto fmt::formatter<MelonDsDs::FormattedPCapFlags>::format(MelonDsDs::Formatted
 
     return formatter<decltype(flagNames)>::format(flagNames, ctx);
 }
+#endif
 
 auto fmt::formatter<MelonDsDs::BiosType>::format(MelonDsDs::BiosType c, format_context& ctx) const -> decltype(ctx.out()) {
     string_view name = "unknown";
diff --git a/src/libretro/net/net.hpp b/src/libretro/net/net.hpp
index 830db63..446353f 100644
--- a/src/libretro/net/net.hpp
+++ b/src/libretro/net/net.hpp
@@ -21,6 +21,17 @@
 
 #ifdef HAVE_NETWORKING_DIRECT_MODE
 #include <Net_PCap.h>
+#else
+namespace melonDS {
+    struct AdapterData {
+        char DeviceName[128];
+        char FriendlyName[128];
+        char Description[128];
+        unsigned char MAC[6];
+        unsigned char IP_v4[4];
+        unsigned int Flags;
+    };
+}
 #endif
 
 #include "Net.h"
diff --git a/cmake/ConfigureFeatures.cmake b/cmake/ConfigureFeatures.cmake
index e0139f1..c10426a 100644
--- a/cmake/ConfigureFeatures.cmake
+++ b/cmake/ConfigureFeatures.cmake
@@ -144,7 +144,7 @@ if (NINTENDO_SWITCH OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "NintendoSwitch"))
     message(STATUS "Building for Nintendo Switch")
 endif()
 
-if (ENABLE_NETWORKING AND (WIN32 OR UNIX) AND HAVE_DYNAMIC)
+if (ENABLE_NETWORKING AND (WIN32 OR UNIX) AND HAVE_DYNAMIC AND NOT ANDROID)
     set(HAVE_NETWORKING_DIRECT_MODE ON)
     message(STATUS "Building with support for direct-mode networking")
 endif()
